app/controllers/nodes_controller.rb

```diff
@@ -16,6 +18,18 @@ def show
     @minion = Minion.find(params[:id])
   end
 
+  def bootstrap
+    salt = Pharos::Salt.new
+    available_roles = %i(master)
+    salt.minions.keys.each do |minion|
+      Minion.transaction do
+        Minion.find_or_create_by!(hostname: minion).assign_role(available_roles)
```

Comment:
It's pass by value, but there is no array cloning here. You are effectively modifying the available_roles. You can check this with a trivial example:

```
def hello
  a = %i(1 2 3)
  hi a
  puts a
end

def hi(a)
  a.delete_at 0
end

hello
```

But still I agree this isn't clear enough. Fixing it.